File: 701215-kadanes-algorithm-GFG/701215-kadanes-algorithm.cpp
class Solution {
  public:
    int maxSubarraySum(vector<int> &arr) {
    int curr = 0;
    int maxx = INT_MIN;
    for(int x : arr){
        curr = max(x,curr+x);
        maxx = max(curr,maxx);
    }
    return maxx;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 701215-kadanes-algorithm-GFG/701215-kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        int maxx=Integer.MIN_VALUE;
        int sum=0;
        int n = arr.length;
        for(int i=0; i<n; i++){
            sum = Math.max(arr[i],sum+arr[i]);
            maxx = Math.max(sum,maxx);
        }
        return maxx;
    }
}


// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna